src/pages/studyView/UserSelections.tsx

```diff
@@ -270,6 +267,81 @@ export default class UserSelections extends React.Component<
             components
         );
 
+        _.reduce(
+            this.props.filter.customDataFilters || [],
+            (acc, customDataFilter) => {
+                const chartMeta = this.props.attributesMetaSet[
+                    customDataFilter.attributeId
+                ];
+                if (chartMeta) {
+                    chartMeta.clinicalAttribute!.datatype;
+                    if (
+                        chartMeta.clinicalAttribute!.datatype ===
+                        DataType.STRING
+                    ) {
+                        // Pie chart filter
+                        acc.push(
+                            <div className={styles.parentGroupLogic}>
```

Comment:
Can you pull this out into its own method, or even its own component? The indentation is making this hard to read.